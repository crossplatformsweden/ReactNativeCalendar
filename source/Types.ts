import { NavigatorTypes } from './navigator/';
import { LoginTypes } from './login/';
import { UtilityTypes } from './utility/';
import { MapTypes, GetLocation, GetMapPointForCoordiantes } from './map';
import {GetSthlmParking, SthlmParkingTypes} from './sthlmparking';
import { StorageTypes, SaveByKey, RemoveKey, GetByKey } from './storage';
import {GeocodeTypes, GeocodeAddress} from './geocode';
import { AutoLogin } from './login/actions/LoginActions';

/**
 * The global application state as contained in Redux
 *
 * All state objects might not be available.
 *
 * @export
 * @interface IApplicationState
 */
export interface IApplicationState {
  route?: NavigatorTypes.INavigationState;
  login?: LoginTypes.ILoginState;
  utility?: UtilityTypes.IUtilityState;
  storage?: StorageTypes.IStorageState;
  map?: MapTypes.IMapState;
  geocoding?: GeocodeTypes.IGeocodeState;
  sthlmparking?: SthlmParkingTypes.ISthlmParkingState;
}

/**
 * Describes possible props for a components.
 *
 * All actions and state objects might not be available.
 *
 * @export
 * @interface IProps
 */
export interface IProps extends IApplicationState {
  // AppLoadingChanged?: appActions.AppLoadingChanged;
  dispatch?: any;
  FacebookLogin?: LoginTypes.FacebookLogin;
  AppErrorChanged?: UtilityTypes.IAppErrorChanged;
  AppLoadingChanged?: UtilityTypes.IAppLoadingChanged;
  GoogleLogin?: LoginTypes.GoogleLogin;
  GeocodeAddress?: GeocodeAddress;
  GetLocation?: GetLocation;
  GetMapPointForCoordiantes?: GetMapPointForCoordiantes;
  GetByKey?: GetByKey;
  RemoveKey?: RemoveKey;
  SaveByKey?: SaveByKey;
  AutoLogin?: AutoLogin;
  GetSthlmParking?: GetSthlmParking;
}
