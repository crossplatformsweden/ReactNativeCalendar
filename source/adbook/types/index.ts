import { Book } from '../actions';
import Moment from 'moment';

export enum ParkingConstants {
    BOOKING_ADDED = 'BOOKING_ADDED',
    BOOKING_CHANGED = 'BOOKING_CHANGED',
    BOOKING_CANCELLED = 'BOOKING_CANCELLED',
}

export type Book = typeof Book;

export enum IVehicleType {
    Van= 'Van',
    Truck= 'Truck',
    Sedan = 'Sedan',
    SUV= 'SUV',
    /**
     * AKA Family Car, SV: Kombi
     *
     * @type {'StationWagon'}
     * @memberof IVehicleType
     */
    StationWagon= 'StationWagon',
}

interface IVehicle {
    vehicleType: IVehicleType;
    vehicleModel?: string;
    regNumber: string;
}

interface IPerson {
    firstName: string;
    lastName: string;
}

interface ITimePeriod {
    timeStart: Moment.Moment;
    timeEnd: Moment.Moment;
}
interface ILocation {
    lat: number;
    lon: number;
}

export interface IBooking {
    timeperiod: ITimePeriod;
    location: ILocation;
    vehicle: IVehicle;
    adress: string;
    lessee: IPerson;
}

export interface IBookingState {
    booking: IBooking;
}

export interface IParkingAction extends IBookingState {
    type: ParkingConstants;
}

export class Booking implements IBooking {
    constructor (readonly timeperiod: ITimePeriod, readonly location: ILocation,
        readonly vehicle: IVehicle, readonly adress: string, readonly lessee: IPerson ) {

        }
}
