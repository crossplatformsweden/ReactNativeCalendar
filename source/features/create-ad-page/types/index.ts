// import { CreateAd } from '../actions';
// export type CreateAd = typeof CreateAd;
import moment from 'moment';

export enum CreateConstants {
    CREATE_SUCCESS = 'CREATE DID SUCCESS',
    CREATE_UPDATED = 'CREATE DID UPDATE',
    CREATE_CANCEL = 'CREATE DID CANCEL',
}

export interface IUser {
    accessToken: string;
    name: string;
    type: string;
  }

export interface ICreateAdPage {
    fromDate: moment.Moment;
    toDate: moment.Moment;
}

export interface ICreateState {
    createAd: ICreateAdPage;
}

export interface ICreateAction extends ICreateState {
    type: CreateConstants;
}