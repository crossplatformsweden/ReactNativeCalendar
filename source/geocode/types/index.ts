// geocode/types/index
/**
 * Geolocation action constant string keys
 * @namespace
 * @public
 * @typedef GeocodingConstants
 * @memberof Constants
 * @type {Object}
 */
export enum GeocodingConstants {
  GEOCODING_BUSY = 'LÃ¤ser kartboken',
  GEOCODING_SUCCESS = 'GEOCODE = SUCCESS',
  GEOCODING_FAILED = 'GEOCODE = FAILED',
}

/**
 * Response from Google API
 * @typedef IGeocodeReducer
 * @type {Object}
 * @memberof TypesModule
 * @public
 * @export
 * @property {Array<IAddressComponent>?} results
 */
export interface IGeocodeState {
  results?: Array<Result>;
}

export interface IGeocodeAction extends IGeocodeState {
  type: GeocodingConstants;
}

// To parse this data:
//
//   import { Convert, Geocode } from "./file";
//
//   const geocode = Convert.toGeocode(json);
//
// These functions will throw an error if the JSON doesn't
// match the expected interface, even if the JSON is valid.

export interface Geocode {
  results?: Result[];
  status?:  string;
}

export interface Result {
  address_components?: AddressComponent[];
  formatted_address?:  string;
  geometry?:           Geometry;
  place_id?:           string;
  types?:              string[];
}

export interface AddressComponent {
  long_name?:  string;
  short_name?: string;
  types?:      string[];
}

export interface Geometry {
  bounds?:        Bounds;
  location?:      Northeast;
  location_type?: string;
  viewport?:      Bounds;
}

export interface Bounds {
  northeast?: Northeast;
  southwest?: Northeast;
}

export interface Northeast {
  lat?: number;
  lng?: number;
}

// Converts JSON strings to/from your types
// and asserts the results of JSON.parse at runtime
export namespace Convert {
  export function toGeocode(json: string): Geocode {
      return cast(JSON.parse(json), o('Geocode'));
  }

  export function geocodeToJson(value: Geocode): string {
      return JSON.stringify(value, null, 2);
  }

  function cast<T>(obj: any, typ: any): T {
      if (!isValid(typ, obj)) {
          throw `Invalid value`;
      }
      return obj;
  }

  function isValid(typ: any, val: any): boolean {
      if (typ === undefined) return true;
      if (typ === null) return val === null || val === undefined;
      return typ.isUnion  ? isValidUnion(typ.typs, val)
              : typ.isArray  ? isValidArray(typ.typ, val)
              : typ.isMap    ? isValidMap(typ.typ, val)
              : typ.isEnum   ? isValidEnum(typ.name, val)
              : typ.isObject ? isValidObject(typ.cls, val)
              :                isValidPrimitive(typ, val);
  }

  function isValidPrimitive(typ: string, val: any) {
      return typeof typ === typeof val;
  }

  function isValidUnion(typs: any[], val: any): boolean {
      // val must validate against one typ in typs
      return typs.find(typ => isValid(typ, val)) !== undefined;
  }

  function isValidEnum(enumName: string, val: any): boolean {
      const cases = typeMap[enumName];
      return cases.indexOf(val) !== -1;
  }

  function isValidArray(typ: any, val: any): boolean {
      // val must be an array with no invalid elements
      return Array.isArray(val) && val.every(element => {
          return isValid(typ, element);
      });
  }

  function isValidMap(typ: any, val: any): boolean {
      if (val === null || typeof val !== 'object' || Array.isArray(val)) return false;
      // all values in the map must be typ
      return Object.keys(val).every(prop => {
          if (!Object.prototype.hasOwnProperty.call(val, prop)) return true;
          return isValid(typ, val[prop]);
      });
  }

  function isValidObject(className: string, val: any): boolean {
      if (val === null || typeof val !== 'object' || Array.isArray(val)) return false;
      let typeRep = typeMap[className];
      return Object.keys(typeRep).every(prop => {
          if (!Object.prototype.hasOwnProperty.call(typeRep, prop)) return true;
          return isValid(typeRep[prop], val[prop]);
      });
  }

  function a(typ: any) {
      return { typ, isArray: true };
  }

  // @ts-ignore
  function e(name: string) {
      return { name, isEnum: true };
  }

  function u(...typs: any[]) {
      return { typs, isUnion: true };
  }

  // @ts-ignore
  function m(typ: any) {
      return { typ, isMap: true };
  }

  function o(className: string) {
      return { cls: className, isObject: true };
  }

  const typeMap: any = {
      'Geocode': {
          results: u(null, a(o('Result'))),
          status: u(null, ''),
      },
      'Result': {
          address_components: u(null, a(o('AddressComponent'))),
          formatted_address: u(null, ''),
          geometry: u(null, o('Geometry')),
          place_id: u(null, ''),
          types: u(null, a('')),
      },
      'AddressComponent': {
          long_name: u(null, ''),
          short_name: u(null, ''),
          types: u(null, a('')),
      },
      'Geometry': {
          bounds: u(null, o('Bounds')),
          location: u(null, o('Northeast')),
          location_type: u(null, ''),
          viewport: u(null, o('Bounds')),
      },
      'Bounds': {
          northeast: u(null, o('Northeast')),
          southwest: u(null, o('Northeast')),
      },
      'Northeast': {
          lat: u(null, 3.14),
          lng: u(null, 3.14),
      },
  };
}
