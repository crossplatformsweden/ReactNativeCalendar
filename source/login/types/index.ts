// login/types/index
import { FacebookLogin, GoogleLogin, BankIdLogin } from '../actions';

export type FacebookLogin = typeof FacebookLogin;

export type GoogleLogin = typeof GoogleLogin;

export type BankIdLogin = typeof BankIdLogin;

/**
 * A user object created when logging in
 * @typedef IUser
 * @type {Object}
 * @public
 * @export
 * @property {string} accessToken - social provider access token
 * @property {string} name - full name of the user
 * @property {string} picture - profile picture url if any
 * @property {string} type - Facebook, Google etc
 * @public
 */
export interface IUser {
  accessToken: string;
  name: string;
  picture?: string;
  type: string;
  username?: string;
  password?: string;
  email?: string;
  autologin?: boolean;
}

export class User implements IUser {
  constructor(
    readonly accessToken: string,
    readonly name: string,
    readonly picture: string = '',
    readonly type: string,
    readonly username: string = null,
    readonly password: string = null,
    readonly email: string = null,
    readonly autologin: boolean = false
  ) {}
}

/**
 * LoginReducer constant string keys
 *
 * @namespace
 * @public
 * @typedef LoginConstants
 * @type {Object}
 */
export enum LoginConstants {
  /**
   * Login failed or was cancelled
   */
  LOGIN_FAILED = 'LOGIN_FAILED',
  /**
   * Login state: authentication succeeded
   */
  LOGIN_SUCCESS = 'LOGIN_SUCCESS',

  /**
   * Login state: authentication in progress
   */
  LOGIN_BUSY = 'Kollar vem du Ã¤r',

  /**
   * Login state: logged out of app
   */
  LOGGED_OUT = 'LOGGED_OUT',
}

export interface ILoginState {
  /**
   * The user payload recived from login if any
   *
   * @type {IUser}
   * @memberof ILoginReducer
   */
  user?: IUser;
  /**
   * Determines if the user is successfully logged in
   *
   * @type {boolean}
   * @memberof ILoginReducer
   */
  isLoggedIn: boolean;
}

export interface ILoginAction extends ILoginState {
  type: LoginConstants;
}
