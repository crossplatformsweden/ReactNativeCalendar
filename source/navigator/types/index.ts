export interface INavigationConstants {
  /**
   * The login screen
   */
  LOGIN?: Function;
  /**
   * The main screen
   */
  MAIN?: Function;
  /**
   *  The logged out screen
   */
  LOGOUT?: Function;
  /**
   * the map screen
   */
  MAP?: Function;
  /**
   * The profile screen
   */
  PROFILE?: Function;
  /**
   * The booking screen
   */
  BOOK?: Function;
}

export interface INavigationState extends INavigationConstants {
  /**
   * Current scene object in navigation
   *
   * @type {object}
   * @memberof INavigationReducer
   */
  scene: object;
}

export interface INavigationAction extends INavigationState {
  type: string;
}

/**
 * Navigation constant keys
 */
export enum NavigationConstants {
  /**
   * The login screen
   */
  LOGIN = 'LoginScreen',
  /**
   * The main screen
   */
  MAIN = 'MainScreen',
  /**
   *  The logged out screen
   */
  LOGOUT = 'LogoutScreen',
  /**
   * the map screen
   */
  MAP = 'MapScreen',
  /**
   * The profile screen
   */
  PROFILE = 'ProfileScreen',
  /**
   * The booking screen
   */
  ADBOOK = 'BookingScreen',
}
