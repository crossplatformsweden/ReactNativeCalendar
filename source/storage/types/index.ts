// storage/types/index

/**
 * App storage Redux action types
 * @type {StorageConstants}
 * @public
 * @type {Object}
 */
export enum StorageConstants {
  STORAGE_GOTKEY = 'STORAGE = GOT KEY',
  STORAGE_REMOVEDKEY = 'STORAGE = REMOVED KEY',
  STORAGE_REMOVED_FAILED = 'STORAGE = REMOVED KEY FAILED',
  STORAGE_NOSUCHKEY = 'STORAGE = NO SUCH KEY',
  STORAGE_SAVE_FAILED = 'STORAGE = SAVE FAILED',
  STORAGE_SAVED = 'STORAGE = SAVED KEY',
  STORAGE_USER_KEY = 'STORAGE KEY = USER',
  STORAGE_BUSY = 'Friskar upp minnet',
}

export interface IKey {
  /**
   * The key in app storage
   *
   * @type {string}
   */
  key: string;
}

/**
 * Reducer properties passed down from StorageReducer
 * @typedef IStorageReducer
 * @type {Object}
 * @public
 * @export
 * @property {Object} value - the app storage value
 * @property {string} key - The app storage key
 */
export interface IStorageState extends IKey {
  /**
   * Value to save / load from app storage
   *
   * @type {object?}
   */
  value?: object;
  type: StorageConstants;
}

/**
 * Creates a {IStorageState} object
 *
 * @param type Storage action type const string
 * @param key Storage key
 * @param value Storage value
 */
export const StorageState = (
  type: StorageConstants = null,
  key: string = null,
  value: object = null
): IStorageState => {
  const r: IStorageState = {
    key,
    value,
    type,
  };

  return r;
};
