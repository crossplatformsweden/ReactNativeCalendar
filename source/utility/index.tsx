// utility/index
import UtilityReducer from './reducers';
import * as UtilityTypes from './types';
import SentryUtility from './SentryUtility';
import BusyIndicator, {
  IBusyIndicatorProps,
  IndicatorType,
  Spinner,
} from './components/BusyIndicator';
import ComponentBase from './components/ComponentBase';
import { AppErrorChanged, AppLoadingChanged } from './actions';

export {
  AppErrorChanged,
  AppLoadingChanged,
  ComponentBase,
  UtilityReducer,
  UtilityTypes,
  BusyIndicator,
  IBusyIndicatorProps,
  IndicatorType,
  Spinner,
  SentryUtility
};
