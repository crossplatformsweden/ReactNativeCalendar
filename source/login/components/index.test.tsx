// login/components/index
/// <reference types="jest"/>
import React from 'react';
import { shallow } from 'enzyme';
import { SocialIcon } from 'react-native-elements';

import { LoginBase } from './';
import { BusyIndicator, UtilityTypes } from '../../utility';
import * as loginTypes from '../types/';
import { StorageTypes } from '../../storage';
import * as types from '../../Types';

jest.unmock('react-native');
jest.unmock('react-native-elements');

// Navigator will call connected components
jest.mock('../../navigator', () => ({
  Routes: 'View',
  AppNavigator: 'Button',
}));

function setup() {
  const user = new loginTypes.User('1', 'test', '', 'Google');

  const props: types.IProps = {
    GetByKey: jest.fn(),
    FacebookLogin: jest.fn(),
    GoogleLogin: jest.fn(),
    AutoLogin: jest.fn(),
    utility: { isBusy: false, busyReason: null, hasError: false },
    login: loginTypes.LoginState(
      loginTypes.LoginConstants.LOGIN_SUCCESS,
      user,
      false
    ),
    storage: StorageTypes.StorageState(),
  };

  const enzymeWrapper = shallow(<LoginBase {...props} />);

  return {
    props,
    enzymeWrapper,
  };
}

describe('components', () => {
  describe('Login', () => {
    it('should render self and subcomponents', () => {
      const { enzymeWrapper } = setup();
      expect(enzymeWrapper).toMatchSnapshot();

      const buttonFacebook = enzymeWrapper
        .find(SocialIcon)
        .at(0)
        .props();
      expect(buttonFacebook.type).toEqual('facebook');

      const buttonGoogle = enzymeWrapper
        .find(SocialIcon)
        .at(1)
        .props();
      expect(buttonGoogle.type).toEqual('google-plus-official');
    });

    it('should be able to press Facebook button', () => {
      const { enzymeWrapper } = setup();
      const buttonFacebook = enzymeWrapper
        .find(SocialIcon)
        .at(0)
        .props();
      buttonFacebook.onPress();
    });

    it('should be able to press Google button', () => {
      const { enzymeWrapper } = setup();
      const buttonGoogle = enzymeWrapper
        .find(SocialIcon)
        .at(1)
        .props();
      buttonGoogle.onPress();
    });
  });
});
