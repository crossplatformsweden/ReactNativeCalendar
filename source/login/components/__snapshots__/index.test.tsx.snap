// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Login should render self and subcomponents 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginBase
    AutoLogin={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    FacebookLogin={[MockFunction]}
    GetByKey={[MockFunction]}
    GoogleLogin={[MockFunction]}
    login={
      Object {
        "isLoggedIn": false,
        "user": Object {
          "accessToken": "1",
          "name": "test",
          "picture": "",
          "type": "Google",
        },
      }
    }
    storage={
      Object {
        "key": "",
        "value": null,
      }
    }
    utility={
      Object {
        "busyReason": null,
        "hasError": false,
        "isBusy": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 120,
                "width": 120,
              }
            }
          />
        </View>,
        <SocialIcon
          button={true}
          iconColor="white"
          iconSize={24}
          id="buttonFacebook"
          onPress={[MockFunction]}
          raised={true}
          title="Sign In With Facebook"
          type="facebook"
        />,
        <SocialIcon
          button={true}
          iconColor="white"
          iconSize={24}
          id="buttonGoogle"
          onPress={[MockFunction]}
          raised={true}
          title="Sign In With Google"
          type="google-plus-official"
        />,
      ],
      "style": Object {
        "flex": 1,
        "marginTop": 10,
        "padding": 20,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Image
            source={1}
            style={
              Object {
                "height": 120,
                "width": 120,
              }
            }
          />,
          "style": Object {
            "alignItems": "flex-start",
            "flexDirection": "row",
            "justifyContent": "center",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "source": 1,
            "style": Object {
              "height": 120,
              "width": 120,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "button": true,
          "iconColor": "white",
          "iconSize": 24,
          "id": "buttonFacebook",
          "onPress": [MockFunction],
          "raised": true,
          "title": "Sign In With Facebook",
          "type": "facebook",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "button": true,
          "iconColor": "white",
          "iconSize": 24,
          "id": "buttonGoogle",
          "onPress": [MockFunction],
          "raised": true,
          "title": "Sign In With Google",
          "type": "google-plus-official",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 120,
                  "width": 120,
                }
              }
            />
          </View>,
          <SocialIcon
            button={true}
            iconColor="white"
            iconSize={24}
            id="buttonFacebook"
            onPress={[MockFunction]}
            raised={true}
            title="Sign In With Facebook"
            type="facebook"
          />,
          <SocialIcon
            button={true}
            iconColor="white"
            iconSize={24}
            id="buttonGoogle"
            onPress={[MockFunction]}
            raised={true}
            title="Sign In With Google"
            type="google-plus-official"
          />,
        ],
        "style": Object {
          "flex": 1,
          "marginTop": 10,
          "padding": 20,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Image
              source={1}
              style={
                Object {
                  "height": 120,
                  "width": 120,
                }
              }
            />,
            "style": Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "justifyContent": "center",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": 1,
              "style": Object {
                "height": 120,
                "width": 120,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "button": true,
            "iconColor": "white",
            "iconSize": 24,
            "id": "buttonFacebook",
            "onPress": [MockFunction],
            "raised": true,
            "title": "Sign In With Facebook",
            "type": "facebook",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "button": true,
            "iconColor": "white",
            "iconSize": 24,
            "id": "buttonGoogle",
            "onPress": [MockFunction],
            "raised": true,
            "title": "Sign In With Google",
            "type": "google-plus-official",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
